internal/controllers/addon/phase_observe_operatorresource.go

```diff
@@ -20,6 +20,45 @@ func (r *olmReconciler) observeOperatorResource(
 	addon *addonsv1alpha1.Addon,
 	csvKey client.ObjectKey,
 ) (requeueResult, error) {
+	if addon == nil {
+		return resultRetry, fmt.Errorf("current addon is nil")
```

Comment:
Golang doesn't have runtime assertions, but this is more of an assertion than a reason to retry. If a `nil` addon was supplied to this method it would be a programmer error and should be fatal.

Reply:
Generally speaking, one nuance of passing pointers is that a nil value can be a valid value to indicate an empty var, and this depends on the assumed intention and programmer preference. I take this review as a nil addon should not even happen, therefore it must fail.

Reply:
Yeah essentially I think it's better to fail loudly here rather than something subtle like a retry because a retry will not resolve the deeper programming error that occurred. Again languages with assertions make it easier to add these checks in a dev build and remove them in prod builds. Since Go doesn't have that I would just recommend something more serious here like a `panic`.